velox/functions/sparksql/tests/StringTest.cpp

```diff
@@ -418,5 +434,85 @@ TEST_F(StringTest, substring) {
   EXPECT_EQ(substring("da\u6570\u636Eta", -3), "\u636Eta");
 }
 
+TEST_F(StringTest, rpad) {
+  std::string invalidString = "Ψ\xFF\xFFΣΓΔA";
+  std::string invalidPadString = "\xFFΨ\xFF";
+
+  // Null arguments
+  EXPECT_EQ(std::nullopt, rpad(std::nullopt, 16, "abc"));
+  EXPECT_EQ(std::nullopt, rpad("xyz", std::nullopt, "abc"));
+  EXPECT_EQ(std::nullopt, rpad("xyz", 16, std::nullopt));
+  // ASCII strings with various values for size and padString
+  EXPECT_EQ("textx", rpad("text", 5, "x"));
+  EXPECT_EQ("text", rpad("text", 4, "x"));
+  EXPECT_EQ("textxy", rpad("text", 6, "xy"));
+  EXPECT_EQ("textxyx", rpad("text", 7, "xy"));
+  EXPECT_EQ("textxyzxy", rpad("text", 9, "xyz"));
+  // Non-ASCII strings with various values for size and padString
+  EXPECT_EQ(
+      "\u4FE1\u5FF5 \u7231 \u5E0C\u671B  \u671B",
+      rpad("\u4FE1\u5FF5 \u7231 \u5E0C\u671B  ", 10, "\u671B"));
+  EXPECT_EQ(
+      "\u4FE1\u5FF5 \u7231 \u5E0C\u671B  \u671B\u671B",
+      rpad("\u4FE1\u5FF5 \u7231 \u5E0C\u671B  ", 11, "\u671B"));
+  EXPECT_EQ(
+      "\u4FE1\u5FF5 \u7231 \u5E0C\u671B  \u5E0C\u671B\u5E0C",
+      rpad("\u4FE1\u5FF5 \u7231 \u5E0C\u671B  ", 12, "\u5E0C\u671B"));
+  EXPECT_EQ(
+      "\u4FE1\u5FF5 \u7231 \u5E0C\u671B  \u5E0C\u671B\u5E0C\u671B",
+      rpad("\u4FE1\u5FF5 \u7231 \u5E0C\u671B  ", 13, "\u5E0C\u671B"));
+  // Empty string
+  EXPECT_EQ("aaa", rpad("", 3, "a"));
+  // Truncating string
+  EXPECT_EQ("", rpad("abc", 0, "e"));
+  EXPECT_EQ("tex", rpad("text", 3, "xy"));
+  EXPECT_EQ(
+      "\u4FE1\u5FF5 \u7231 ",
+      rpad("\u4FE1\u5FF5 \u7231 \u5E0C\u671B  ", 5, "\u671B"));
+  // Invalid UTF-8 chars
+  EXPECT_EQ(invalidString + "x", rpad(invalidString, 8, "x"));
+  EXPECT_EQ("abc" + invalidPadString, rpad("abc", 6, invalidPadString));
+}
+
+TEST_F(StringTest, lpad) {
+  std::string invalidString = "Ψ\xFF\xFFΣΓΔA";
+  std::string invalidPadString = "\xFFΨ\xFF";
+
+  // Null arguments
+  EXPECT_EQ(std::nullopt, lpad(std::nullopt, 16, "abc"));
+  EXPECT_EQ(std::nullopt, lpad("xyz", std::nullopt, "abc"));
+  EXPECT_EQ(std::nullopt, lpad("xyz", 16, std::nullopt));
+  // ASCII strings with various values for size and padString
```

Comment:
These tests are very dense and hard to read. Consider adding empty lines between sections, e.g.

```
  EXPECT_EQ("xyzxytext", lpad("text", 9, "xyz"));

  // Non-ASCII strings with various values for size and padString
  EXPECT_EQ(
```

Please, make sure all comments are full sentences.